File: msgraph-mail/dotnet/Users/DateTimeTimeZone.cs
using Kiota.Abstractions.Serialization;
using System;
using System.Collections.Generic;
using System.Linq;
namespace Graphdotnetv4.Users {
    public class DateTimeTimeZone : IParsable<DateTimeTimeZone> {
        /// <summary>Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.</summary>
        public IDictionary<string, object> AdditionalData { get; private set; } = new Dictionary<string, object>();
        /// <summary>A single point of time in a combined date and time representation ({date}T{time}; for example, 2017-08-29T04:00:00.0000000).</summary>
        public string DateTime { get; set; }
        /// <summary>The serialization information for the current model</summary>
        public IDictionary<string, Action<DateTimeTimeZone, IParseNode>> DeserializeFields => new Dictionary<string, Action<DateTimeTimeZone, IParseNode>> {
            {
                "dateTime", (o,n) => { o.DateTime = n.GetStringValue(); }
            },
            {
                "timeZone", (o,n) => { o.TimeZone = n.GetStringValue(); }
            },
        };
        /// <summary>Represents a time zone, for example, 'Pacific Standard Time'. See below for more possible values.</summary>
        public string TimeZone { get; set; }
        /// <summary>
        /// Serialiazes information the current object
        /// <param name="writer">Serialization writer to use to serialize this model</param>
        /// </summary>
        public void Serialize(ISerializationWriter writer) {
            writer.WriteStringValue("dateTime", DateTime);
            writer.WriteStringValue("timeZone", TimeZone);
            writer.WriteAdditionalData(AdditionalData);
        }
    }
}

File: msgraph-mail/typescript/src/users/dateTimeTimeZone.ts
import {SerializationWriter, ParseNode, Parsable} from '@microsoft/kiota-abstractions';

export class DateTimeTimeZone implements Parsable<DateTimeTimeZone> {
    /** Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.  */
    public readonly additionalData: Map<string, unknown> = new Map<string, unknown>();
    /** A single point of time in a combined date and time representation ({date}T{time}). For example, '2019-04-16T09:00:00'.  */
    public dateTime?: string | undefined;
    /** Represents a time zone, for example, 'Pacific Standard Time'. See below for possible values.  */
    public timeZone?: string | undefined;
    /**
     * The deserialization information for the current model
     * @returns a Map<string, (item: DateTimeTimeZone, node: ParseNode) => void>
     */
    public getFieldDeserializers () : Map<string, (item: DateTimeTimeZone, node: ParseNode) => void> {
        return new Map<string, (item: DateTimeTimeZone, node: ParseNode) => void>([
            ["dateTime", (o, n) => { o.dateTime = n.getStringValue(); }],
            ["timeZone", (o, n) => { o.timeZone = n.getStringValue(); }],
        ]);
    };
    /**
     * Serialiazes information the current object
     * @param writer Serialization writer to use to serialize this model
     * @returns a void
     */
    public serialize (writer: SerializationWriter) : void {
        writer.writeStringValue("dateTime", this.dateTime);
        writer.writeStringValue("timeZone", this.timeZone);
        writer.writeAdditionalData(this.additionalData);
    };
}
